src/index.tsx

```diff
@@ -1,10 +1,43 @@
-import ReactDOM from 'react-dom';
-import { HashRouter } from 'react-router-dom';
+import { createRoot } from 'react-dom/client';
+import { Navigate, RouterProvider, createHashRouter } from 'react-router-dom';
+import React from 'react';
 import { App } from './App';
+import { HomePage } from './routes/HomePage';
+import { TabsPage } from './routes/TabsPage';
+import { NotFoundPage } from './routes/NotFoundPage';
 
-ReactDOM.render(
-  <HashRouter>
-    <App />
-  </HashRouter>,
-  document.getElementById('root'),
+const router = createHashRouter([
+  {
+    path: '/',
+    element: <App />,
+    children: [
+      {
+        index: true,
+        element: <HomePage />,
+      },
+      {
+        path: 'tabs',
+        element: <TabsPage />,
+        children: [
+          {
+            path: ':tabId',
+          },
+        ],
+      },
+      {
+        path: 'home',
+        element: <Navigate to="/" replace />,
+      },
+      {
+        path: '*',
+        element: <NotFoundPage />,
+      },
+    ],
+  },
+]);
+
+createRoot(document.getElementById('root') as HTMLDivElement).render(
+  <React.StrictMode>
+    <RouterProvider router={router} />
+  </React.StrictMode>,
 );
```

Comment:
It's not the correct way to structure your application.
1. Let's use `App` as a root component, not as a layout
2. Create `AppRouter` that will render routing logic in your `app` component 
3. Create `Layout` component where you can render your view:
4. Remove the container to components to wrap them not in a react fragment

```
    <NavBar />

    <div className="section">
      <Outlet/>
    </div>
```

```suggestion
createRoot(document.getElementById('root') as HTMLDivElement).render(
  <React.StrictMode>
    <RouterProvider router={router} />
  </React.StrictMode>,
);
```